tests/test_gmail.py

```diff
@@ -1,59 +1,100 @@
 import os.path
+import pickle
+from copy import copy
 from datetime import datetime
 
 import mock
 from dateutil.tz import tzutc
+from google.oauth2.credentials import Credentials
+from mock import patch
+from six.moves import configparser
 
 import bugwarrior.services.gmail as gmail
-from .base import ServiceTest, AbstractServiceTest
+from bugwarrior.config import ServiceConfig
+from bugwarrior.services.gmail import GmailService
+
+from .base import AbstractServiceTest, ConfigTest, ServiceTest
+
+CREDENTIAL_DATA = {
+    "token": "itsatokeneveryone",
+    "refresh_token": "itsarefreshtokeneveryone",
+    "token_uri": "https://oauth2.googleapis.com/token",
+    "client_id": "example.apps.googleusercontent.com",
+    "client_secret": "itsasecrettoeveryone",
+    "scopes": ["https://www.googleapis.com/auth/gmail.readonly"],
+}
+
+
+class TestGmailService(ConfigTest):
+    def setUp(self):
+        super(TestGmailService, self).setUp()
+        self.config = configparser.RawConfigParser()
+        self.config.add_section("general")
+        self.config.add_section("myservice")
+
+        mock_data = mock.Mock()
+        mock_data.path = self.tempdir
+        self.config.data = mock_data
+
+        self.service_config = ServiceConfig(GmailService.CONFIG_PREFIX, self.config, "myservice")
+
+    def test_get_credentials_exists(self):
+        mock_api = mock.Mock()
+        gmail.GmailService.build_api = mock_api
+        service = GmailService(self.config, "general", "myservice")
+
+        expected = Credentials(**copy(CREDENTIAL_DATA))
+        with open(service.credentials_path, "wb") as token:
+            pickle.dump(expected, token)
+
+        self.assertEqual(service.get_credentials().to_json(), expected.to_json())
+
+    def test_get_credentials_expired(self):
```

Comment:
Could you add some comment(s) explaining how this test works? It isn't obvious why mocking `Credentials.refresh` to return `None` would yield the same results as `expected.__dict__["expiry"] = datetime.now()`.

Reply:
I'm new to mocking and agree my initial test wasn't very clear, but after sleeping on it I think I have a cleaner/more self explanatory solution for this via 7e4c263. Let me know what you think!

Reply:
Looks good, thanks!